test/ecto/query/builder/update_test.exs

```diff
@@ -19,9 +19,19 @@ defmodule Ecto.Query.Builder.UpdateTest do
       [compile] = query.updates
       assert compile.expr == [set: [foo: {:==, [], [{{:., [], [{:&, [], [0]}, :foo]}, [], []}, {:^, [], [0]}]}]]
       assert compile.params == [{1, {0, :foo}}]
+
+      query = "foo" |> update([p], set: %{foo: p.foo == ^1})
+      [compile] = query.updates
+      assert compile.expr == [set: [foo: {:==, [], [{{:., [], [{:&, [], [0]}, :foo]}, [], []}, {:^, [], [0]}]}]]
+      assert compile.params == [{1, {0, :foo}}]
+
+      query = "foo" |> update([p], set: %{"foo" => p.foo == ^1})
+      [compile] = query.updates
+      assert compile.expr == [set: [{"foo", {:==, [], [{{:., [], [{:&, [], [0]}, :foo]}, [], []}, {:^, [], [0]}]}}]]
+      assert compile.params == [{1, {0, :foo}}]
     end
 
-    test "raises on non-keyword lists" do
+    test "raises on non keyword/string-key lists" do
```

Comment:
I'm not sure here … First of all personally I would avoid making such change in case you are not updating anything else (here test body).

You are removing `-` character, but using it in other part. Maybe I'm wrong, but it looks like a small inconsistency (?) … Shouldn't it be `non keyword/string key` or `non-keyword/string-key` instead? Personally I would name it like: `non keyword or string key`.

Also I do not see a `string key` usage. There are only two checks:
1. If `List` which is not `Keyword`
2. If `:set` option is `List` which is not `Keyword`
Did you forgot to add an extra check here?